File: kattis/classy.java
import java.util.*;

class Main {
	public static long calculateScore(String classes) {
		String[] lst = classes.split("-");
		long counter = 0;
		long multiplier = 100000000000L;
		int curr = lst.length;
		while (curr > 0) {
			if (lst[curr - 1].equals("middle")) {
				counter += multiplier * 2;
			} else if (lst[curr - 1].equals("upper")) {
				counter += multiplier * 5;
			} else {
				counter += multiplier;
			}

			multiplier /= 10;
			curr--;
		}
		curr = 10 - lst.length;
		while (curr > 0) {
			counter += multiplier * 2;
			curr--;
			multiplier /= 10;
		}
		return -counter;
	}
	public static void main(String[] args) {
		Kattio io = new Kattio(System.in);
		int tc = io.getInt();
		for (int i = 0; i < tc; i++) {
			int ln = io.getInt();
			List<String[]> lst = new ArrayList<String[]>();
			for (int j = 0; j < ln; j++) {
				String name = io.getWord();
				String class_train = io.getWord();
				String dummy = io.getWord();
				name = name.substring(0, name.length() - 1);
				lst.add(new String[]{name, class_train});
			}
			Collections.sort(lst, new Comparator<String[]>() {
				public int compare(String[] a, String[] b) {
					return a[0].compareTo(b[0]);
				}
			});

			Collections.sort(lst, new Comparator<String[]>() {
				public int compare(String[] a, String[] b) {
					return calculateScore(a[1]).compareTo(calculateScore(b[1]));
				}
			});

			for (String[] data : lst) {
				io.println(data[0]);
				io.println(calculateScore(data[1]));
			}
			io.println("==============================");
		}
		io.flush();
	}
}

File: kattis/classy.py
def main():
    testcase = input()
    for _ in range (int(testcase)):
        n = int(input())
        lst = []
        for _ in range(n):
            name, kelas, gjls = input().split()
            name = name[:-1]
            kelas = kelas.split('-')
            multiplier = 10**12
            panjang = len(kelas)
            kelas = (10-panjang)*['middle'] + kelas
            score = 0
            panjang = 10
            while panjang > 0:
                if kelas[panjang-1] == 'upper':
                    score += 5 * multiplier
                elif kelas[panjang-1] == 'middle':
                    score += 2 * multiplier
                else:
                    score += multiplier
                multiplier /= 10
                panjang -= 1
            lst.append([-score, name])
        lst.sort()
        for i in range (len(lst)):
            print(lst[i])
        print("==============================")


if __name__ == '__main__':
    main()